{
  "url": "https://api.github.com/repos/ethereum/go-ethereum/issues/26165",
  "repository_url": "https://api.github.com/repos/ethereum/go-ethereum",
  "labels_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/26165/labels{/name}",
  "comments_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/26165/comments",
  "events_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/26165/events",
  "html_url": "https://github.com/ethereum/go-ethereum/issues/26165",
  "id": 1446022355,
  "node_id": "I_kwDOAOvK985WMIzT",
  "number": 26165,
  "title": "IPv6 network support in nat extip doesn't work correctly",
  "user": {
    "login": "ritave",
    "id": 1614945,
    "node_id": "MDQ6VXNlcjE2MTQ5NDU=",
    "avatar_url": "https://avatars.githubusercontent.com/u/1614945?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/ritave",
    "html_url": "https://github.com/ritave",
    "followers_url": "https://api.github.com/users/ritave/followers",
    "following_url": "https://api.github.com/users/ritave/following{/other_user}",
    "gists_url": "https://api.github.com/users/ritave/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/ritave/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/ritave/subscriptions",
    "organizations_url": "https://api.github.com/users/ritave/orgs",
    "repos_url": "https://api.github.com/users/ritave/repos",
    "events_url": "https://api.github.com/users/ritave/events{/privacy}",
    "received_events_url": "https://api.github.com/users/ritave/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [
    {
      "id": 72233650,
      "node_id": "MDU6TGFiZWw3MjIzMzY1MA==",
      "url": "https://api.github.com/repos/ethereum/go-ethereum/labels/type:bug",
      "name": "type:bug",
      "color": "FF5E5E",
      "default": false,
      "description": ""
    }
  ],
  "state": "open",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 2,
  "created_at": "2022-11-11T21:30:43Z",
  "updated_at": "2022-11-15T23:17:44Z",
  "closed_at": null,
  "author_association": "NONE",
  "active_lock_reason": null,
  "body": "My router assigns me both a local IPv4 address and a public IPv6 one. There's no ability to forwards ports, and so I want to use the IPv6 address as my node location.\r\n\r\nThat address is publicly accessible as tested using ssh from external location. Yet when I ask geth to use it, it doesn't work as expected and can't connect to any peers.\r\n\r\nAddress in this issue is randomized.\r\n\r\n#### System information\r\n\r\nGeth version: `geth version`\r\n```\r\nGeth\r\nVersion: 1.10.25-stable\r\nArchitecture: amd64\r\nGo Version: go1.19.3\r\nOperating System: linux\r\nGOPATH=\r\nGOROOT=\r\n```\r\n\r\nCL client & version: None\r\nOS & Version: `Linux 6.0.7 NixOS from 22.05.4055.fa842715565-x86_64`\r\n\r\n\r\n#### Expected behaviour\r\n\r\n`geth -nat=extip:ff16:90c4:d2b0:fec4:c16b:e1c5:d4ae:46e2` results in IPv6 address used for connections.\r\n\r\n\r\n#### Actual behaviour\r\n\r\n`geth -nat=extip:ff16:90c4:d2b0:fec4:c16b:e1c5:d4ae:46e2` set both `admin.nodeInfo.ip` and `admin.nodeInfo.enode` default to `127.0.0.1` address\r\n",
  "closed_by": null,
  "reactions": {
    "url": "https://api.github.com/repos/ethereum/go-ethereum/issues/26165/reactions",
    "total_count": 0,
    "+1": 0,
    "-1": 0,
    "laugh": 0,
    "hooray": 0,
    "confused": 0,
    "heart": 0,
    "rocket": 0,
    "eyes": 0
  },
  "timeline_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/26165/timeline",
  "performed_via_github_app": null,
  "state_reason": null
}
[
  {
    "url": "https://api.github.com/repos/ethereum/go-ethereum/issues/comments/1312606771",
    "html_url": "https://github.com/ethereum/go-ethereum/issues/26165#issuecomment-1312606771",
    "issue_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/26165",
    "id": 1312606771,
    "node_id": "IC_kwDOAOvK985OPMoz",
    "user": {
      "login": "ritave",
      "id": 1614945,
      "node_id": "MDQ6VXNlcjE2MTQ5NDU=",
      "avatar_url": "https://avatars.githubusercontent.com/u/1614945?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/ritave",
      "html_url": "https://github.com/ritave",
      "followers_url": "https://api.github.com/users/ritave/followers",
      "following_url": "https://api.github.com/users/ritave/following{/other_user}",
      "gists_url": "https://api.github.com/users/ritave/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/ritave/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/ritave/subscriptions",
      "organizations_url": "https://api.github.com/users/ritave/orgs",
      "repos_url": "https://api.github.com/users/ritave/repos",
      "events_url": "https://api.github.com/users/ritave/events{/privacy}",
      "received_events_url": "https://api.github.com/users/ritave/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2022-11-13T00:52:52Z",
    "updated_at": "2022-11-13T00:52:52Z",
    "author_association": "NONE",
    "body": "Ps. I confirm the port 30303 is open and visible from outside.",
    "reactions": {
      "url": "https://api.github.com/repos/ethereum/go-ethereum/issues/comments/1312606771/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/ethereum/go-ethereum/issues/comments/1316000039",
    "html_url": "https://github.com/ethereum/go-ethereum/issues/26165#issuecomment-1316000039",
    "issue_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/26165",
    "id": 1316000039,
    "node_id": "IC_kwDOAOvK985OcJEn",
    "user": {
      "login": "ritave",
      "id": 1614945,
      "node_id": "MDQ6VXNlcjE2MTQ5NDU=",
      "avatar_url": "https://avatars.githubusercontent.com/u/1614945?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/ritave",
      "html_url": "https://github.com/ritave",
      "followers_url": "https://api.github.com/users/ritave/followers",
      "following_url": "https://api.github.com/users/ritave/following{/other_user}",
      "gists_url": "https://api.github.com/users/ritave/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/ritave/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/ritave/subscriptions",
      "organizations_url": "https://api.github.com/users/ritave/orgs",
      "repos_url": "https://api.github.com/users/ritave/repos",
      "events_url": "https://api.github.com/users/ritave/events{/privacy}",
      "received_events_url": "https://api.github.com/users/ritave/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2022-11-15T23:17:44Z",
    "updated_at": "2022-11-15T23:17:44Z",
    "author_association": "NONE",
    "body": "Ok, the issue was that geth can't sync historic blocks without consensus client. With both working in tandem it works perfectly.\r\n\r\nThe current Ethereum launchpad doesn't reflect those requirements - my assumption was that I could run geth to sync while I set-up the rest of the system. I'd appreciate more clear documentation / errors that geth won't do anything without a consensus client.",
    "reactions": {
      "url": "https://api.github.com/repos/ethereum/go-ethereum/issues/comments/1316000039/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  }
]
